CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# What this repository is
+
+This repository contains sources of Compose Multiplatform supporting projects (Gradle plugin, Resources, samples, templates, etc.) which are used in [Compose Multiplatform](https://github.com/JetBrains/compose-multiplatform).
+Compose Multiplatform project provides support only for non-Android targets.
+
+# What this repository is not
+
+The main codebase for Compose Multiplatform is located in a [separate repository](https://github.com/JetBrains/compose-multiplatform-core),
+which is a fork of [AOSP](https://github.com/androidx/androidx).
+Android-specific contributions should be submitted in AndroidX repository.
+
+# Contributing
+
+We love contributions!
+If you want to find some issues to start off with,
+try [this query](https://youtrack.jetbrains.com/issues/CMP?q=tag:%20%7BUp%20For%20Grabs%7D%20%23Unresolved) which should find all open Compose Multiplatform issues that are marked as "up-for-grabs".
+
+If you'd like to fix a bug or add a feature, [create a YouTrack issue](https://youtrack.jetbrains.com/newIssue?project=CMP) for it first.
+You'll then be able to discuss the proposed change with maintainers.
+It would help to avoid situations when the change conflicts with some other feature or help discover potential edge cases.
+
+Currently only committers can assign issues to themselves so just add a comment if you're starting work on it.
+
+Make sure to follow the [development guide](https://github.com/JetBrains/compose-multiplatform-core/blob/jb-main/MULTIPLATFORM.md) in the core repository for local set up.
+
+## Submitting patches
+
+The best way to submit a patch is to [fork the project on GitHub](https://help.github.com/articles/fork-a-repo/) and then send us a
+[pull request](https://help.github.com/articles/creating-a-pull-request/) to the `master` branch via [GitHub](https://github.com).
+
+If you create your own fork, it might help to enable rebase by default
+when you pull by executing
+``` bash
+git config --global pull.rebase true
+```
+This will avoid your local repo having too many merge commits
+which will help keep your pull request simple and easy to apply.
+
+## Rules for commit messages
+
+Most of these rules are originated from the [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+article, and it's highly recommended to read it.
+
+### Rules on commit messages' content
+
+1. Use the body to explain what and why vs. how
+    * Please make an extra effort to explain why changes are needed for every non-trivial modification.
```

Comment:
```suggestion
    * Please make an extra effort to explain why changes are needed for every non-trivial modification.
    * Describe how you tested your changes
    * Add Release Notes
    * Follow the [PR template](https://github.com/JetBrains/compose-multiplatform/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
```

Reply:
PR template is used automatically in a new PR, is it needed to mention it here explicitly?

Reply:
If someone reads the guide, then creates a PR, it can be a surprise if they already prepared a message and now they need to adapt it again by following additional rules. Mentioning it before creating a PR adds awareness about the PR format.

In the proposed change, the link just reiterates what is described in the guide. We could also change it differently - just provide the link and remove the rest ("explain why changes are needed", "Describe how you tested", ...).